File: Ballast.Core/src/Server/Messaging/Events/SignIn/PlayerSignedInEvent.cs
using Ballast.Core.Models;
using System;

namespace Ballast.Core.Messaging.Events.SignIn
{
    public class PlayerSignedInEvent : EventBase 
    {

        public override string Id => nameof(PlayerSignedInEvent);

        public IPlayer Player { get; private set; } 
        public DateTime TimeStamp { get; private set; }

        public PlayerSignedInEvent(Player player) : base() {
            Player = player;
            TimeStamp = DateTime.UtcNow;
        }

    }
}

File: Ballast.Core/src/Client/messaging/events/sign-in/player-signed-in-event.ts
import { IPlayer } from '../../../models/player';
import { getUtcNow } from '../../../utility/date-helpers';
import { EventBase } from '../../event-base';

export class PlayerSignedInEvent extends EventBase {

    public static readonly id: Symbol = Symbol.for('PlayerSignedInEvent');

    public get id() {
        return PlayerSignedInEvent.id;
    }

    public readonly player?: IPlayer; 
    public readonly timeStamp?: Date;

    public constructor(player?: IPlayer) {
        super();
        this.player = player;
        this.timeStamp = getUtcNow();
    }

}